core/src/main/scala/org/http4s/multipart/Multipart.scala

```diff
@@ -1,14 +1,29 @@
 package org.http4s
 package multipart
 
+import cats.implicits._
 import org.http4s.headers._
 
 final case class Multipart[F[_]](parts: Vector[Part[F]], boundary: Boundary = Boundary.create) {
   def headers: Headers =
     Headers(
       List(
-        `Transfer-Encoding`(TransferCoding.chunked),
+        contentLength
+          .flatMap(len => `Content-Length`.fromLong(len).toOption)
+          .getOrElse(`Transfer-Encoding`(TransferCoding.chunked)),
         `Content-Type`(MediaType.multipartType("form-data", Some(boundary.value)))
       )
     )
+
+  def contentLength: Option[Long] = {
+    val newlineLength = 2L
+    parts.traverse(_.contentLength).map { lengths =>
```

Comment:
Where are you accounting for the parts' headers' lengths?

Reply:
D'oh! Any pointers on the easiest way to get the length of a header? :)

Reply:
Yes, #2622!  Go ahead and cherry-pick that if you get back to this before it's approved.